shared/common-adapters/profile-card.tsx

```diff
@@ -0,0 +1,243 @@
+          position="top center"
+        >
+          <Kb.Icon fontSize={14} type={Platforms.serviceIdToIcon(serviceId)} />
+        </Kb.WithTooltip>
+      ))}
+      {!expanded && serviceIds.length > 4 && (
+        <Kb.ClickableBox onClick={() => setExpanded(true)} style={styles.expand}>
+          <Kb.Meta title={`+${serviceIds.length - 3}`} backgroundColor={Styles.globalColors.greyDark} />
+        </Kb.ClickableBox>
+      )}
+    </Kb.Box2>
+  )
+}
+
+const ProfileCard = ({clickToProfile, showClose, username}: Props) => {
+  const userDetails = Container.useSelector(state => Tracker2Constants.getDetails(state, username))
+  const followThem = Container.useSelector(state => Tracker2Constants.followThem(state, username))
+  const followsYou = Container.useSelector(state => Tracker2Constants.followsYou(state, username))
+  const isSelf = Container.useSelector(state => state.config.username === username)
+
+  const dispatch = Container.useDispatch()
+  React.useEffect(() => {
+    dispatch(Tracker2Gen.createShowUser({asTracker: false, skipNav: true, username}))
+  }, [dispatch, username])
+  const _changeFollow = React.useCallback(
+    (follow: boolean) => dispatch(Tracker2Gen.createChangeFollow({follow, guiID: userDetails.guiID})),
+    [dispatch, userDetails]
+  )
+
+  return (
+    <Kb.Box2 direction="vertical" style={styles.container} alignItems="center">
+      {showClose && (
+        <Kb.Icon type="iconfont-close" onClick={() => {}} boxStyle={styles.close} padding="tiny" />
+      )}
+      <Kb.ConnectedNameWithIcon
+        onClick={clickToProfile && 'profile'}
+        colorFollowing={true}
+        username={username}
+        metaStyle={styles.connectedNameWithIconMetaStyle}
+        metaTwo={userDetails.fullname || ''}
+        withProfileCardPopup={false}
+      />
+      {userDetails.state === 'checking' ? (
+        <Kb.ProgressIndicator type="Large" />
+      ) : (
+        <>
+          <ServiceIcons userDetails={userDetails} />
+          {userDetails.bio && (
+            <Kb.Text type="Body" center={true}>
+              {userDetails.bio}
+            </Kb.Text>
+          )}
+        </>
+      )}
+      {!isSelf &&
+        (followThem ? (
+          <FollowButton
+            key="unfollow"
+            following={true}
+            onUnfollow={() => _changeFollow(false)}
+            waitingKey={Tracker2Constants.waitingKey}
+            style={styles.button}
+          />
+        ) : (
+          <FollowButton
+            key="follow"
+            following={false}
+            followsYou={followsYou}
+            onFollow={() => _changeFollow(true)}
+            waitingKey={Tracker2Constants.waitingKey}
+            style={styles.button}
+          />
+        ))}
+    </Kb.Box2>
+  )
+}
+
+type WithProfileCardPopupProps = {
+  username: string
+  children: (ref: React.Ref<any>) => React.ReactNode
+}
+
+export const WithProfileCardPopup = ({username, children}: WithProfileCardPopupProps) => {
```

Comment:
should add a comment  why we're doing this

Reply:
You mean for import cycle? I added one in usernames component in the latest feedback.